src/providers/ruby.rs

```diff
@@ -211,6 +215,14 @@ impl RubyProvider {
         }
         Ok(false)
     }
+
+    fn uses_gem_dep(&self, app: &App, dep: &str) -> bool {
+        if app.includes_file("Gemfile") {
```

Comment:
@Milo123459 should we also potentially be locking for a lockfile to detect nondirect dependencies like we do in the Node Provider?